diff --git a/dub.sdl b/dub.sdl
index 3589cb4..ac15195 100644
--- a/dub.sdl
+++ b/dub.sdl
@@ -52,6 +52,8 @@ configuration "linux-full" {
 	versions "InBranding"
 	
 	dependency "dportals" version="~>0.1.0"
+	subConfiguration "lumars" "lua51-dynamic"
+	versions "LUA_51"
 }
 
 configuration "osx-full" {
diff --git a/source/nijiexpose/plugins/package.d b/source/nijiexpose/plugins/package.d
index 11e9a10..2766930 100644
--- a/source/nijiexpose/plugins/package.d
+++ b/source/nijiexpose/plugins/package.d
@@ -34,13 +34,11 @@ Plugin[] insPlugins;
     Initializes Lua support
 */
 void insLuaInit() {
-    // LuaSupport support = loadLua();
-
-    // if (support == LuaSupport.noLibrary || support == LuaSupport.badLibrary) {
-    //     couldLoadLua = false;
-    //     insLogWarn("Could not load Lua support...");
-    // } else insLogInfo("Lua support initialized.");
-    insLogInfo("Lua support initialized. (Statically linked for now)");
+    LuaSupport support = loadLua("libluajit-5.1.so.2");
+    if (support == LuaSupport.noLibrary || support == LuaSupport.badLibrary) {
+        couldLoadLua = false;
+        insLogWarn("Could not load Lua support...");
+    } else insLogInfo("Lua support initialized.");
 
     // Create Lua state
     state = new LuaState(luaL_newstate());
@@ -55,6 +53,7 @@ void insLuaInit() {
 
 void insLuaUnload() {
     lua_close(state.handle());
+    unloadLua();
     destroy(state);
 }
 
