diff --git a/dub.sdl b/dub.sdl
index 69166c7..a09cf44 100644
--- a/dub.sdl
+++ b/dub.sdl
@@ -3,7 +3,7 @@ description "nijigenerate UI Library"
 authors "seagetch"
 copyright "Copyright © 2024, seagetch"
 license "BSD 2-clause"
-dependency "nijilive" version="~>0.0.1"
+dependency "nijilive" version="~>0.0.0"
 dependency "tinyfiledialogs" version="~>0.10.1"
 dependency "i2d-opengl" version="~>1.0.0"
 dependency "bindbc-sdl" version="~>1.1.2"
